Manual/ModuleSystem.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2025 Lean FRO LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Sebastian Ullrich
+-/
+
+import VersoManual
+import Manual.Meta
+
+open Verso.Genre Manual
+open Verso.Genre.Manual.InlineLean
+
+#doc (Manual) "The Module System" =>
+%%%
+%%%
+
+The module system is an experimental Lean 4 feature allowing for finer-grained control over what information is exported from Lean modules and imported into other modules, improving average build times, executable sizes, API design options, and more.
+
+The module system is activated by prefixing a Lean file with the `module` keyword.
+At the time of writing, it is additionally guarded by the `experimenta
+.module` option that has to be set to `true` in the project's Lakefile.
+`module`s can only import other `module`s so adoption has to be done top-down.
+Non-`module`s can import `module`s and will ignore all module system-specific annotations.
+
+# Visibility
+
+The main distinction the module system introduces is between the *public scope* that contains all information visible to other modules via `import` and the *private scope* that is not imported by default.
+Both declarations and imports themselves are scoped in this way.
+
+The default scope is private.
+The new modifier `public` before a declaration or import puts it into the public scope instead.
+No information from the private scope can be used in the public scope to ensure information in the latter still makes sense when only it is imported into other modules.
+
+:::TODO
+These examples should be output-checked and elaborated with `-Dexperimental.module=true`
+:::
+
+```
+module
+
+def priv : Nat := 0
+
+public theorem priv_eq_priv : priv = priv := rfl
+```
+
+`public section` can be used to switch the default scope for declarations, with `private` locally negating it.
+This is mostly intended to ease porting and avoiding merge conflicts during it.
+
+Marking a declaration as public at minimum makes its "signature", i.e. its name and type, visible.
+Some specific declarations/terms may still put other parts in the private scope as follows:
+* `by` used in the public scope on a proposition puts the resulting proof in the private scope (by wrapping it in a public helper theorem).
+* `def` puts its body in the private scope by default, thus it cannot be unfolded when used in the public scope.
```

Comment:
```suggestion
* `def` puts its body in the private scope by default. The defined constant cannot be unfolded when used in the public scope.
```
Comma splice. Also, "it" should refer to the defined constant, but here it sounded more like the keyword/syntactic for `def` itself.